src/modules/hip/kernel/spatter.hpp

```diff
@@ -0,0 +1,322 @@
+#include <hip/hip_runtime.h>
+#include "hip/rpp_hip_common.hpp"
+#include "func_specific/spatter_mask.hpp"
+#include <random>
+
+__device__ void spatter_hip_compute(uchar *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, d_float8 *mask_f8, d_float8 *maskInv_f8, float4 *pix_f4)
```

Comment:
can't you avoid different definitions using Template argument for the srcPtr?

Reply:
Templating could be used if every arithmetic inside the 4 functions are exactly the same except for the data-type, which can be templated as <T>. But in this case, we have different computations for each function.